python/tests/engine/test_python_spark_transformation_functions.py

```diff
@@ -508,27 +508,28 @@ def test_apply_plus_one_double(self, mocker):
 
         expected_schema = StructType(
             [
-                StructField("col_0", DoubleType(), True),
+                StructField("col_0", IntegerType(), True),
                 StructField("col_1", StringType(), True),
                 StructField("col_2", BooleanType(), True),
             ]
         )
         expected_df = pd.DataFrame(
             data={
-                "col_0": [2.0, 3.0],
+                "col_0": [2, 3],
                 "col_1": ["test_1", "test_2"],
                 "col_2": [True, False],
             }
         )
         expected_spark_df = spark_engine._spark_session.createDataFrame(
             expected_df, schema=expected_schema
         )
+        spark_df = spark_engine._spark_session.createDataFrame(df, schema=schema)
 
         # Arrange
         def tf_fun(a) -> int:
             return a + 1
 
-        td = self._create_training_dataset(tf_fun, "double")
+        td = self._create_training_dataset(tf_fun, int)
```

Comment:
Sure, if you change the test it will pass :) Now it's essentially the identical to `test_apply_plus_one_int`

But what the original test showed is that spark does not seem to allow implicit casts from the udf results (int) to the specified output type (double), which leads to the column being filled with nullls. The python engine, however, is less strict and allows this conversion. Probably hat means we have to make the python engine "less smart", or we just accept the fact and document the difference.

Reply:
Is resolved in https://github.com/logicalclocks/feature-store-api/pull/795